src/main/java/com/epam/aidial/core/controller/PublicationController.java

```diff
@@ -132,10 +168,19 @@ private ResourceDescription decodePublication(String path) {
         return resource;
     }
 
-    private void checkAccess(ResourceDescription resource) {
-        String bucket = BlobStorageUtil.buildInitiatorBucket(context);
+    private void checkAccess(ResourceDescription resource, boolean onlyAdmin, boolean onlyPrivate) {
```

Comment:
This method require java doc or proper naming.
This 2 flags are controversial. I expect either admin access required or user access. In your case you have both true.

`
checkAccess(ResourceDescription resource, boolean requireAdminAccess);

// if user has admin access - immediately return 
if (accessService.hasAdminAccess(context)) {
return;
}

// if private resource - check bucket
if (!requireAdminAccess && resource.getBucketLocation().equals(bucket)) {
 return;
}

throw new FobiddenException();
`

Reply:
refactored